File: app/src/main/java/pers/cxd/commonmodule/presenters/RxPresenter.java
package pers.cxd.commonmodule.presenters;

import io.reactivex.rxjava3.disposables.CompositeDisposable;
import pers.cxd.corelibrary.mvp.Presenter;

public class RxPresenter<V, M> extends Presenter<V, M> {

    protected CompositeDisposable mSubscription;

    @Override
    public void attach(V v, M m) {
        super.attach(v, m);
        mSubscription = new CompositeDisposable();
    }

    @Override
    public void detach() {
        super.detach();
        if (mSubscription != null){
            mSubscription.dispose();
        }
        mSubscription = null;
    }

    @Override
    public boolean notDetach() {
        return super.notDetach() && mSubscription != null;
    }

}

File: app/src/main/java/pers/cxd/commonmodule/presenters/RxPresenter.kt
package pers.cxd.commonmodule.presenters

import io.reactivex.rxjava3.disposables.CompositeDisposable
import pers.cxd.corelibrary.base.BaseView
import pers.cxd.corelibrary.base.Presenter


open class RxPresenter<V : BaseView, M : Any> : Presenter<V, M>() {
    @JvmField
    protected var mSubscription: CompositeDisposable? = null
    override fun attach(v: V, m: M) {
        super.attach(v, m)
        mSubscription = CompositeDisposable()
    }

    override fun detach() {
        super.detach()
        mSubscription?.dispose()
        mSubscription = null
    }

    public override fun notDetach(): Boolean {
        return super.notDetach() && mSubscription != null
    }
}